### DISCUSSION TITLE
Grid Situation for v3.0

### DISCUSSION THREAD

**Discussion by james-rae** [fgpv-vpgf/fgpv-vpgf#4103] [Team Posts]
See brief issue thread here first: https://github.com/fgpv-vpgf/fgpv-vpgf/issues/3299

### Fun Facts

- The "basic grid" is not in good working shape
- The enhanced grid is a large file
- Most sites will likely use the enhanced grid (or an alternative enhanced grid)

### Current Approach

Issue thread is suggesting making our enhanced grid the default grid for `v3.0`.

PROs:

- Quickly fixes up all sample pages
- Will allow an enhanced grid to be used without needing any plugin registration (which is probably the most common deployment)

CONs:

- If in `v3.1` we make the basic grid the default, then any sites upgrading will need to add the enhanced grid plugin references to their page (or accept the grid downgrade).
- If someone in `v3.0` wants to use their own grid plugin, the fancy grid will still be bundled into their download.  Probably an unlikely scenario.

### Random Idea

- Is there a way to structure the code bundle so that enhanced grid is "default", but does not get downloaded if a different plugin overrides it?

**Comment by james-rae** [comment]
Another idea:  we keep the broken "basic grid" as the default in RAMP, and for `v3.0` add a `console.warn` to inform any would-be developers that "using the basic grid is a real bad idea, and we suggest using the enhanced grid until `v3.1` is released".   Warning is suppressed if any grid plugin overrides the simple grid.

This allows us to keep the large fancy grid package out of the core bundle. It also removes any type of page updates when `v3.1` comes out (with the exception of person who wants to use basic grid).

Major con is it requires all the sample pages to be updated now to use the enhanced grid (hoping this is just a multi-file replace in notepad++)

**Comment by mweech** [comment]
We should not release the basic grid as default.  i expect we will get more issues with individuals being surprised by the grid being different then we would providing some upgrade steps for v3.1. in most cases, eliminating the grid from the bundle is just delaying the inevitable loading of the same library.
